### PR TITLE
feat: add localization to bigcommerce example

### PR DESCRIPTION
**Pull request by joshuawootonn**

### CODE REVIEW

File: examples/bigcommerce/components/cart.tsx [hunk 1]
```diff
@@ -21,19 +23,26 @@ type CartLineItemProps = {
...
       <a className={`flex w-full space-x-4 items-end`} key={lineItem.id}>
-        <img src={lineItem.image_url} alt={`Product image for ${lineItem.name}`} width={70} />
+        <img
+          src={lineItem.image_url}
+          alt={`Product image for ${product.name ?? lineItem.name}`}
+          width={70}
+        />
         <div className="flex flex-col flex-grow items-start justify-between">
-          <div className="text-base text-black">{lineItem.name}</div>
+          <div className="text-base text-black">{product.name ?? lineItem.name}</div>
```
Comment (joshuawootonn) [inline examples/bigcommerce/components/cart.tsx:39]: There is no api for getting the Metafields of a lineitem, so I am preferring the `product.name` to the `lineItem.name` here.

File: examples/bigcommerce/components/cart.tsx [hunk 2]
```diff
@@ -202,7 +212,8 @@ export function ProductAddToCartButton({ className }: ProductAddToCartButtonProp
   const [quantity, setQuantity] = useState(1)
   const [addToCartState, setAddToCartState] = useState<AddToCartState>('initial')
   const { addItem } = useCart()
-  const product = useProduct()
+  const product = useProductFromPath()
```
Comment (joshuawootonn) [inline examples/bigcommerce/components/cart.tsx:215]: this hook was renamed to make way for `getProduct(id: number)`

File: examples/bigcommerce/lib/bigcommerce/graphql.ts [hunk 3]
```diff
@@ -30,11 +30,20 @@ export const PRODUCT_FRAGMENT = /* GraphQL */ `
...
+          value
+        }
+      }
+    }
   }
 `
 
 export const PRODUCTS_QUERY = /* GraphQL */ `
-  query Products {
+  query Products($hasLocale: Boolean = false, $locale: String = "null") {
```
Comment (joshuawootonn) [inline examples/bigcommerce/lib/bigcommerce/graphql.ts:46]: this is just a rip off of how commerce is handling this

**Comment by migueloller** [comment]
> [feat: update to test if it fixes the revalidate issue](https://github.com/makeswift/makeswift/pull/269/commits/f9b3a27e1e1c1dc78e85bbc484e498facb9463cc)

So did it? Feels like this commit message is a temporary one?

**Comment by joshuawootonn** [comment]
> > [feat: update to test if it fixes the revalidate issue](https://github.com/makeswift/makeswift/pull/269/commits/f9b3a27e1e1c1dc78e85bbc484e498facb9463cc)
> 
> So did it? Feels like this commit message is a temporary one?

checking now

**Comment by joshuawootonn** [comment]
> > [feat: update to test if it fixes the revalidate issue](https://github.com/makeswift/makeswift/pull/269/commits/f9b3a27e1e1c1dc78e85bbc484e498facb9463cc)
> 
> So did it? Feels like this commit message is a temporary one?

It didn't work. The changes in this commit are temp and will be fixed up

File: examples/bigcommerce/components/cart.tsx [hunk 4]
```diff
@@ -21,19 +23,26 @@ type CartLineItemProps = {
...
         <div className="flex flex-col flex-grow items-start justify-between">
-          <div className="text-base text-black">{lineItem.name}</div>
+          <div className="text-base text-black">{product?.name ?? lineItem.name}</div>
           <div className="text-sm text-green mb-2">{formatPrice(lineItem.list_price)}</div>
           <div className="flex justify-center items-center space-x-2">
             <button
               disabled={cartState === 'loading'}
               className="disabled:cursor-not-allowed"
-              aria-label="Decrease quantity"
+              aria-label={t('decrease-quantity') ?? 'Decrease quantity'}
```
[the remark this answers is not in the document]
Comment (migueloller) [reply examples/bigcommerce/components/cart.tsx:45]: We seem to be using a fallback in some places and not in others. I think we should be consistent.

File: examples/bigcommerce/components/cart.tsx (hunk 2 above)
Comment (migueloller) [reply examples/bigcommerce/components/cart.tsx:215]: I'm not sure I understand this comment. How does renaming `useProduct` to `useProductFromPath` make way for a function `getProduct` that's not a hook?

File: examples/bigcommerce/public/locales/es/cart.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "add-to-cart": "Añadir al carrito",
+  "add-to-cart-confirming": "Adicional",
+  "decrease-quantity": "Disminuir la cantidad",
+  "remove-from-cart": "Retirar",
+  "increase-quantity": "Aumentar la cantidad",
```
Comment (migueloller) [inline examples/bigcommerce/public/locales/es/cart.json:6]: This should be `Aumentar cantidad`

File: examples/bigcommerce/public/locales/es/cart.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  "add-to-cart": "Añadir al carrito",
```
Comment (migueloller) [inline examples/bigcommerce/public/locales/es/cart.json:2]: This should be `Agregar al carrito`

File: examples/bigcommerce/components/cart.tsx (hunk 2 above)
Comment (joshuawootonn) [reply examples/bigcommerce/components/cart.tsx:215]: Sorry I meant `getProduct(id: number)`

File: examples/bigcommerce/README.md [hunk 7]
```diff
@@ -105,6 +105,81 @@ Note: the env for this example corresponds to the [Vercel Commerce BigCommerce e
...
+  - This determines what APIs have access to this `metafield`. Since we query product data from the storefront API, we will need the `read_and_sf_access` `permission_set`.
+- `namespace`
+  - This indicates the locale that this `metafield` belongs to. In our situation, it should match the identifier this translation belongs to.
+  - A translation to Spanish would go under the `es` namespace.
+- `key`
+  - The property in our product that this translation corresponds to.
+- `value`
+  - The translated text itself.
+
+Here is an example of I how we translated the values in our plant store. Monstera is Monstruoso in Spanish.
```
[the remark this answers is not in the document]
Comment (joshuawootonn) [reply examples/bigcommerce/README.md:138]: Changed this to using "Blue Lily" and "Lirio Azul" for the example

File: examples/bigcommerce/components/cart.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (joshuawootonn) [reply examples/bigcommerce/components/cart.tsx:45]: Yeah. I agree in theory. The type changed when I upgraded and it was throwing a warning that null was not assignable to undefined. A good compromise is just being explicit about the generic type when it is inferring the wrong thing -> `t<string>('decrease-quantity')`.

File: examples/bigcommerce/components/cart.tsx (hunk 2 above)
Comment (migueloller) [reply examples/bigcommerce/components/cart.tsx:215]: You just wrote exactly what you wrote in your original comment. Do you mean `useProduct(id: number)`?

File: examples/bigcommerce/components/cart.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (migueloller) [reply examples/bigcommerce/components/cart.tsx]: Yeah, either that or using a non-null assertion. This is really a TypeScript issue so it feels weird so solve it by adding code that affects the running JS

File: examples/bigcommerce/next-i18next.config.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  i18n: {
+    defaultLocale: 'en-US',
+    locales: ['en-US', 'es'],
+  },
+  localePath:
```
Comment (migueloller) [inline examples/bigcommerce/next-i18next.config.js:6]: Did this fix the issue???

**Review by migueloller — APPROVED** [review]

File: examples/bigcommerce/components/cart.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (migueloller) [reply examples/bigcommerce/components/cart.tsx]: Your solution looks good

File: examples/bigcommerce/next-i18next.config.js (hunk 8 above)
Comment (migueloller) [reply examples/bigcommerce/next-i18next.config.js:6]: That's awesome! So glad it was a quick fix!

Comment (joshuawootonn) [reply examples/bigcommerce/next-i18next.config.js:6]: Me too. I will create a PR for this in the POC tomorrow!

**Pull request merged**